dts/bindings/sensor/zephyr,adc-thermistor.yaml

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2023, Basalte bv
+# SPDX-License-Identifier: Apache-2.0
+
+compatible: "zephyr,adc-thermistor"
+
+include: sensor-device.yaml
+
+description: |
+  General Purpose ADC based thermistor sensor with an optional lookup table.
```

Comment:
Please, add the MUST about  lookup table even size in the devicetree documentation. User should be notified about restrictions in the documentation.